src/Components/Patient/PatientConsentRecords.tsx

```diff
@@ -37,41 +42,69 @@ export default function PatientConsentRecords(props: {
 
   const fileUpload = useFileUpload({
     type: "CONSENT_RECORD",
+    allowedExtensions: ["pdf", "jpg", "jpeg", "png"],
   });
 
   const fileManager = useFileManager({
     type: "CONSENT_RECORD",
     onArchive: async () => {
       refetch();
+      refetchFiles();
+      refetchArchivedFiles();
+    },
+    onEdit: async () => {
+      refetch();
+      refetchFiles();
+      refetchArchivedFiles();
     },
   });
 
+  const [refreshTrigger, refetch] = useRefresh();
+
   const { data: patient } = useQuery(routes.getPatient, {
     pathParams: {
       id: patientId,
     },
   });
-  const {
-    data: consultation,
-    refetch,
-    loading,
-  } = useQuery(routes.getConsultation, {
-    pathParams: { id: consultationId! },
-    onResponse: (data) => {
-      if (data.data && data.data.consent_records) {
-        setConsentRecords(data.data.consent_records);
-      }
-    },
-  });
 
   const [showDeleteConsent, setShowDeleteConsent] = useState<string | null>(
     null,
   );
 
-  const [consentRecords, setConsentRecords] = useState<ConsentRecord[] | null>(
-    null,
+  const [consentRecords, setConsentRecords] = useState<
+    PatientConsentModel[] | null
+  >(null);
+
+  const { data: unarchivedFiles, refetch: refetchFiles } = useQuery(
+    routes.viewUpload,
+    {
+      query: {
+        file_type: "CONSENT_RECORD",
+        associating_id: consentRecords?.map((cr) => cr.id).join(","),
+        limit: 1000,
```

Comment:
was this intensional?

Reply:
Yes, this was intentional because we currently don't have any other way to handle paginated queries apart from the pagination component.. but it doesn't fit the use case here 😅. Need something like react-query's useInfiniteQuery hook, which we can take up later on

Reply:
Sure, create an issue now, maybe it can be picked up later.

cc: @nihal467